### ISSUE TITLE
Per-observation attempts

### ISSUE THREAD

**Issue by janper** [subdgtl/Monoceros#27]
Implement optional:
1. make observation
2. try to canonicalize the world
3. if ok, repeat 1.
4. if contradictory, roll back 1. and try a different attempt

This behaves like the step solver (interrupted and again executed) - the solution does not completely change if contradictory.
This gives different results and has to be evaluated if they are nice.

**Comment by yanchith** [comment]
So this is to prevent the world changing completely if the step solver runs into a dead end? Seems viable and completely implementable in Monoceros - the WFC core already gives us all the tools to do this. The only thing to keep in mind is having a reliable strategy for changing the random state.

**Comment by janper** [comment]
I know, that's why the issue is in the Monoceros repo.
I'm only worried the result won't be so nice and the performance will be quite bad.

**Comment by janper** [comment]
> So this is to prevent the world changing completely if the step solver runs into a dead end? 
Yes. Also to harmonize the behavior of the step-solver and its counterpart in GH.
